hydra-node/test/Hydra/Logging/SignalingSpec.hs

```diff
@@ -0,0 +1,30 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Hydra.Logging.SignalingSpec where
+
+import Hydra.Prelude
+import Test.Hydra.Prelude
+
+import Control.Concurrent.Class.MonadSTM (modifyTVar', newTVarIO, readTVarIO)
+import Control.Monad.IOSim (runSimOrThrow)
+import Hydra.API.Server (Server (..))
+import Hydra.API.ServerOutput (ServerOutput (SomeHeadInitializing))
+import Hydra.Cardano.Api (Tx)
+import Hydra.Chain (HeadId (..))
+import Hydra.Chain.Direct.Handlers (DirectChainLog (SomeHeadObserved))
+import Hydra.Logging (nullTracer, traceWith)
+import Hydra.Logging.Messages (HydraLog (DirectChain))
+import Hydra.Logging.Signaling (installSignal, withSignaling)
+
+spec :: Spec
+spec =
+  it "signals observations from the chain" $ do
+    let someHeadId = HeadId "1234"
+        signalled = runSimOrThrow $ withSignaling nullTracer $ \tracer signalVar -> do
+          sentSignals <- newTVarIO []
+          let server = Server @Tx $ \output -> atomically (modifyTVar' sentSignals (output :))
+          installSignal signalVar server
+          traceWith tracer (DirectChain $ SomeHeadObserved someHeadId ["12345"])
+          readTVarIO sentSignals
+
+    signalled `shouldBe` [SomeHeadInitializing someHeadId ["3132333435"]]
```

Comment:
I'm a bit surprised that this passes? Reading this, I would expect the two singleton arrays to be equal `["12345"]`?

Reply:
No, because we start from some text and then we base16-encode it to send to the client.